src/attributes/initializer.js

```diff
@@ -0,0 +1,35 @@
+const natives = Object.assign({}, {
+  getValue: function(attr) {
+    return attr.default;
+  },
+  shouldInitialize: function(attr) {
+    return typeof attr.default !== 'function';
+  }
+});
+
+const functions = Object.assign({}, {
+  getValue: function(attr, instance) {
+    return attr.default(instance);
+  },
+  shouldInitialize: function(attr) {
+    return typeof attr.default === 'function';
```

Comment:
What do you think about using Lodash's [`isFunction`](https://lodash.com/docs/4.17.5#isFunction) since we're already using that for other types? Example: https://github.com/talyssonoc/structure/blob/75e92f3591ec3e8936a28628d75fe36b0546e4e6/src/coercion/number.js#L1

Reply:
Also, we should use shortcut syntax here (I thought we had this rule in the linter, may be worth to add it in a separate PR). Like this:

```js
Object.assign({}, {
  getValue(attr, instance) { ... }
// ...
```